editor/editor_properties_array_dict.cpp

```diff
@@ -587,6 +603,28 @@ bool EditorPropertyArray::_is_drop_valid(const Dictionary &p_drag_data) const {
 		return true;
 	}
 
+	if (drop_type == "resource") {
+		Ref<Resource> res = drag_data["resource"];
+		if (!res.is_valid()) {
+			return false;
+		}
+
+		String res_type = res->get_class();
+		StringName script_class;
+		if (res->get_script()) {
+			script_class = EditorNode::get_singleton()->get_object_custom_type_name(res->get_script());
+		}
+
+		for (int j = 0; j < allowed_type.get_slice_count(","); j++) {
```

Comment:
This loop 😬
I know it's copied, but you should do `split()` and iterate the string array instead.